ts/features/cieLogin/components/CieIdLoginWebView.tsx

```diff
@@ -0,0 +1,168 @@
+  const loginUri = getCieIDLoginUri(spidLevel, isUat);
+
+  const navigateToCieIdAuthenticationError = useCallback(() => {
+    navigation.navigate(ROUTES.AUTHENTICATION, {
+      screen: ROUTES.AUTHENTICATION_CIE_ID_ERROR
+    });
+  }, [navigation]);
+
+  const { shouldBlockUrlNavigationWhileCheckingLollipop, webviewSource } =
+    useLollipopLoginSource(navigateToCieIdAuthenticationError, loginUri);
+
+  useEffect(() => {
+    // https://reactnative.dev/docs/linking#open-links-and-deep-links-universal-links
+    const urlListenerSubscription = Linking.addEventListener(
+      "url",
+      ({ url }) => {
+        // if the url is of this format: iologincie:https://idserver.servizicie.interno.gov.it/idp/login/livello2mobile?value=e1s2
+        // extract the part after iologincie: and dispatch the action to handle the login
+        if (url.startsWith("iologincie:")) {
+          const [, continueUrl] = url.split("iologincie:");
+
+          if (continueUrl) {
+            // https://idserver.servizicie.interno.gov.it/cieiderror?cieid_error_message=Operazione_annullata_dall'utente
+            // We check if the continueUrl is an error
+            if (continueUrl.indexOf("cieiderror") !== -1) {
+              // And we extract the error message and show it in an alert
+              const [, errorMessage] = continueUrl.split(
+                "cieid_error_message="
+              );
+              // TODO: remove this after https://pagopa.atlassian.net/browse/IOPID-2322
+              Alert.alert("Login error ❌", errorMessage ?? "error");
+            } else {
+              setAuthenticatedUrl(continueUrl);
+            }
+          }
+        }
+      }
+    );
+
+    return () => urlListenerSubscription.remove();
+  }, []);
+
+  const handleOnShouldStartLoadWithRequest = (
+    event: WebViewNavigation
+    // eslint-disable-next-line sonarjs/cognitive-complexity
+  ): boolean => {
+    const url = event.url;
+
+    if (shouldBlockUrlNavigationWhileCheckingLollipop(url)) {
+      return false;
+    }
+
+    if (url.indexOf("token=") !== -1) {
+      const [, token] = url.split("token=");
+      if (token) {
+        // show success alert with dismiss button navigatin back
+        dispatch(loginSuccess({ token: token as SessionToken, idp: "cie" })); // ! the idp is cie?
+      }
+      return false;
+    }
+
+    if (
+      url.indexOf("livello1") >= 0 || // SpidL1
+      (url.indexOf("livello2") >= 0 && url.indexOf("livello2mobile") === -1) || // SpidL2
+      url.indexOf("nextUrl") >= 0 || // SpidL3 iOS
+      url.indexOf("openApp") >= 0 // SpidL3 Android
+    ) {
+      if (Platform.OS === "ios") {
+        const urlForCieId = `CIEID://${url}&sourceApp=iologincie`;
+        Linking.openURL(urlForCieId).catch(navigateToCieIdAuthenticationError);
+      } else {
+        openCieIdApp(
+          url,
+          result => {
+            if (result.id === "ERROR") {
+              navigateToCieIdAuthenticationError();
```

Comment:
Here we should discriminate by `code`. More specifically we know that here we can have only `CIEID_OPERATION_CANCEL` according to that [document](https://pagopa.atlassian.net/wiki/spaces/IAEI/pages/1305084137/Analisi+della+gestione+degli+errori+CieID+L2+e+L3) and you should navigate to [this screen](https://www.figma.com/design/hfNTDuzZ0QEiVZVuT3UB7d/Errori-in-app?node-id=657-2496&t=nXSwniVp6SeF5DbR-4).

You can add a comment like you did for iOS 

`// TODO: remove this after https://pagopa.atlassian.net/browse/IOPID-2322`

and take this into account in the next JIRA ticket.